mne/annotations.py

```diff
@@ -567,6 +567,50 @@ def crop(self, tmin=None, tmax=None, emit_warning=False, verbose=None):
 
         return self
 
+    @verbose
+    def rename(self, mapping, verbose=None):
+        """Rename annotation description(s). Operates inplace.
+
+        Parameters
+        ----------
+        mapping : dict
+            A dictionary mapping the old description to a new description
+            name e.g. {'1.0' : 'Control', '2.0' : 'Stimulus'}.
+        %(verbose_meth)s
+
+        Returns
+        -------
+        self : mne.Annotations
+            The modified Annotations object.
+
+        Notes
+        -----
+        .. versionadded:: 0.24.0
+        """
+        self.description = _rename_annotations(self.description,
+                                               mapping, verbose)
+        return self
+
+
+@verbose
+def _rename_annotations(desc, mapping, verbose=None):
+
+    if isinstance(mapping, dict):
```

Comment:
this `if isinstance...  else ValueError` pattern could probably be replaced by our `_validate_types` helper.  If you're not up for that refactor, at the very least this should raise a `TypeError` instead.